src/cmd/promremotebench/write.go

```diff
@@ -184,10 +210,14 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
 		wg.Add(1)
 		go func() {
+			metrics := c.metrics[url]
```

Comment:
This seems like it may be a bit racy? Could you capture `metrics := c.metrics[url]` outside of the go func instead, that seems like it should do it?